sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/jdbc/JdbcPropertiesBeanPostProcessor.java

```diff
@@ -78,17 +85,38 @@ public Object postProcessBeforeInitialization(Object bean, String beanName) thro
                 return bean;
             }
 
+
             try {
                 Map<String, String> enhancedProperties = buildEnhancedProperties(databaseType, properties);
-                String enhancedUrl = connectionString.enhanceConnectionString(enhancedProperties);
-                ((DataSourceProperties) bean).setUrl(enhancedUrl);
+                connectionString.enhanceProperties(enhancedProperties);
+                enhanceUserAgent(databaseType, connectionString);
+                ((DataSourceProperties) bean).setUrl(connectionString.getJdbcUrl());
```

Comment:
can we refactor to something like

```
// from------
 enhanceUserAgent(databaseType, connectionString);

// to------

connectionString.enhanceUserAgent();
```

Reply:
Not a good idea to couple the UA with the connection string. I've refactored the code to introduce a `JdbcConnectionStringEnhancer` class.